src/util/tempdir.cpp

```diff
@@ -107,11 +115,13 @@ temp_working_dirt::temp_working_dirt(const std::string &name_template):
 {
   old_working_directory=get_current_working_directory();
   if(chdir(path.c_str())!=0)
-    CHECK_RETURN(false);
+    throw system_exceptiont("Couldn't change directory to " + path);
 }
 
-temp_working_dirt::~temp_working_dirt()
+temp_working_dirt::~temp_working_dirt() noexcept(false)
 {
+  // FIXME throwing from destructors is generally a bad idea
+  //       but I can't think of anything else that'd be reasonable here?
```

Comment:
I think this code just shouldn't be doing this. If changing into the directory is desired, then the call site should do so. As a matter of fact, this code seems unused...

Reply:
Would you mind creating a fresh PR that just removes this code? I agree it doesn't fit this PR, but having said PR merged before this one would make things a lot easier.

Reply:
@tautschnig https://github.com/diffblue/cbmc/pull/3054